problems/removeStringsEndingInBang.js

```diff
@@ -11,8 +11,12 @@
 * returns: ["", "a"]
 */
 
-function removeStringsEndingInBang() {
+function removeStringsEndingInBang(arr) {
+    for(let i = 0; i <= arr.length; i++) {
+        arr = arr.filter(i => i !== "!")
```

Comment:
very close, this is a better use for filter, but the only thing we would need to check in this filter condition is to see if the last character of the string is a bang operator `!`